contracts/05-tamagotchi-auto/src/lib.rs

```diff
@@ -127,6 +148,92 @@ async fn main() {
             msg::reply(TmgEvent::AttributeBought(attribute_id), 0)
                 .expect("Error in a reply `TmgEvent::AttributeBought`");
         }
+        TmgAction::CheckState => {
+            let payload;
+
+            update_stats(
+                tamagotchi,
+                tamagotchi.fed_block,
+                tamagotchi.entertained_block,
+                tamagotchi.slept_block,
+            );
+
+            if tamagotchi.fed <= HUNGER_THRESHOLD {
+                payload = TmgEvent::FeedMe;
+            } else if tamagotchi.entertained <= BOREDOM_THRESHOLD {
+                payload = TmgEvent::PlayWithMe;
+            } else if tamagotchi.slept <= ENERGY_THRESHOLD {
+                payload = TmgEvent::WantToSleep;
+            } else {
+                payload = TmgEvent::NothingToDo;
+            }
+
+            if exec::program_id() != msg::source() {
+                msg::reply(payload, 0).expect("Error in reply");
+                return;
+            }
+
+            if tamagotchi.reservations.len() == 1 {
+                *can_send_delayed_message = false;
+                *contract_send_a_delayed_message = false;
+                let Some(reservation_id) = tamagotchi.reservations.pop() else {
+                    panic!("Error getting reservation id");
+                };
+
+                msg::send_delayed_from_reservation(
+                    reservation_id,
+                    tamagotchi.owner,
+                    TmgEvent::MakeReservation,
+                    0,
+                    DELAY_OF_ONE_MINUTE,
+                )
+                .expect("Error sending message from reservation");
+            } else {
+                let Some(reservation_id) = tamagotchi.reservations.pop() else {
+                    panic!("Error getting reservation id");
+                };
+
+                msg::send_delayed_from_reservation(
+                    reservation_id,
+                    exec::program_id(),
+                    TmgAction::CheckState,
+                    0,
+                    DELAY_OF_ONE_MINUTE,
+                )
+                .expect("Error sending message from reservation");
```

Comment:
you can reserve a large amount of gas for sending more than one delayed message
It's possible to check the the remaining gas and if it becomes low, take the next reservation